SU-HSDO/suhumsci
# STN-336 :  QA Feedback Sprint 15

#  🛑 [STN-336](https://sparkbox.atlassian.net/browse/STN-336) DO NOT MERGE - SEE new PR #592

## Summary
Implement fixes for items discovered during during the Sprint 15 QA Feedback session.

The work in this PR has been browser tested. ✅ 

## Need Review By (Date)
5/6/2020

## Urgency
low

## Steps to Test
1. In the CLI, run `npm run test` and confirm all our Sass tests pass.
2. Confirm each of the following items have been resolved.


- [x] External Link icon is inconsistently applying on WYSIWYG
    * This was resolved in QA Sprint 14 Feedback PR. Confirm by visiting the QA Real Econ Content page which was built using the WYSIWYG. http://economics.suhumsci.loc/qa/qa-real-econ-content
<br>

- [x] Prevent external links (and external link styling) from appearing in the Drupal Administration Toolbar
    * I was unable to duplicate this issue locally, but you can see it on the dev environment for archaeology.
    * In the CLI, run `lando drush @archaeology.dev uli` to create a login URL for the archaeology site. Copy and paste the URL provided into your web browser. Eg. `http://archaeology-dev.stanford.edu/user/reset/1/1588259656/aG-T70_eSMOy5wezcVEagQHUBraFVzBZiOXu_5Vwdrk/login` 
    * Confirm the external link icon is gone (as a result of the work in QA Feedback Sprint 14)
  * _hmm I still see something (an arrow) on interior pages_
    * Open the inspector tools. Find the anchor link for the `Subscription active` tab. Confirm that setting the `background-image` to `none` will resolve the external link styling issue. ✅ 
    * The database settings which exclude external links from displaying in particular locations have also been updated to include the toolbar administration.
<br>

- [x] External Link icon applying to postcards and breaking arrow link styling
    * NOTE: The styles causing this bug were removed in the Sprint 14 QA Feedback PR
    * To confirm, go to http://economics.suhumsci.loc/qa/qa-cards
    * Click on a contextual link to edit the page
    * Edit a postcard
    * Update the READ MORE LINK to an external link such as `http://google.com` and Save
<br>

- [x] Email links apply envelope on postcards
    * Go to http://economics.suhumsci.loc/qa/qa-cards
    * Click on a contextual link to edit the page
    * Edit a postcard
    * Update the READ MORE LINK to a mail link such as `mailto: abc@xyz.com` and Save
    * Confirm the link on the postcard does not have an email envelope icon to the right of the arrow icon
<br>

- [x] Email links in views aren't styled appropriately
    * Go to http://economics.suhumsci.loc/people/faculty
    * Confirm email icon is present, the text and icon are the right color, and the icon changes color when hovering over the link text and the icon _hmm I dont see the icon color changing_. 
    * You can also see what a link with an email icon looks like within a sentence on this page: http://economics.suhumsci.loc/seminars-workshops/list-seminars. Look at the fourth item of `Development` in the list.
<br>

- [x] Exposed Filter in Sidebar
    * Go to http://economics.suhumsci.loc/courses
    * Click on `Layout` in the Drupal Admin bar below the site navigation
    * In the left hand sidebar of `Section 2`, click `+ Add block`
    * Under `Courses (Views)` select `Custom Econ Courses: All List Exposed Form` and `Add block`
    * Save the layout and go back to the courses page
<br>

- [x] Full pager Styling
    * Go to http://economics.suhumsci.loc/about/news
    * Scroll to the bottom of the page. The full pager style should already be implemented.
    * Look at the `active` and `hover` state background circle. It should look like a circle when there is only 1digit and like a button when there are 2 or more digits.
<br>

- [x] Mini pager Styling
    * Go to http://economics.suhumsci.loc/about/news
    * Click on the contextual link to edit the `View`
    * Under `Block Setting` find `Pager`, under `Use pager`, click on `Full`
    * Update to `Paged output, mini pager` then Apply and Save
<br>

- [x] Image floating/wrapping in text areas
    * Create a basic page and add text areas that include headings, images floated to the right and left, ordered and unordered lists, and body copy
    * Verify the following issues have been resolved:
        * When an image is floated to the right alongside a list, the bullets align beside the list as expected.
        * When an image floats left or right in a text area component, the content in a component placed below does not wrap around the floated image. (Ensure floats are cleared after each row.)
        * In the text area editor, make sure you are always able to `edit` an image. There was an issue with not being able to click on the `edit` button for an image when it was floated alongside lists.
<br>

- [x] Event Series Menu Block
    * Go to http://economics.suhumsci.loc/seminars-workshops/list-seminars and to http://economics.suhumsci.loc/events/public-economics-and-environmental-economics
    * Verify the sidebar on the Public Economics and Environmental Economics page is the same width as the sidebar on the List of Seminars page
<br>

- [x] View Titles get an odd small styling
    * Go to http://economics.suhumsci.loc/seminars-workshops/upcoming-seminars-and-events
    * I _think_ this may have been resolved in the work on headings in the QA Feedback for Sprint 14. Take a look and see if you agree. 👍 
<br>


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
